WeatherApp/WeatherApp.Webpage/Services/CurrentWeatherService.cs

```diff
@@ -9,20 +10,21 @@ namespace WeatherApp.WebSite.Services
 {
     public class CurrentWeatherService : ICurrentWeatherService
     {
-        const string API_KEY = "3c850b0463346d2fffad82b66d5eb561";
+        readonly string _apiKey;
+        readonly string _baseUrl;
 
-        public CurrentWeatherService(IWebHostEnvironment webHostEnvironment)
+        public CurrentWeatherService(IConfiguration configuration)
         {
-            WebHostEnvironment = webHostEnvironment;
+            _apiKey = configuration.GetValue<string>("ApiKeys:WeatherForecast");
+            _baseUrl = configuration.GetValue<string>("ApiBaseUrls:CurrentWeather");
         }
 
-        public IWebHostEnvironment WebHostEnvironment { get; }
-
         public CurrentWeather GetCurrentWeather(string city)
         {
-            string jsonString = "";
-            string url = $"https://api.openweathermap.org/data/2.5/weather?q={city}&appid={API_KEY}&units=metric";
+            string urlParameters = $"appid={_apiKey}&q={city}&units=metric";
+            string url = _baseUrl + urlParameters;
 
+            string jsonString = "";
             using (var client = new HttpClient())

```

Comment:
You should use `IHttpClientFactory` instead of creating your own httpclient.  In this form it makes hard to test this method.
On the other hand it is a hidden dependency. From an outer perspective the caller doesn't know that this service using http calls.
Please examine [this article](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/http-requests?view=aspnetcore-5.0) and use `IHttpClientFactory` instead.